src/Pratt.elm

```diff
@@ -215,31 +215,29 @@ Also note that `oneOf` and `andThenOneOf` parsers may call recursively
 
 -}
 expressionHelp : Config state e -> Int -> e -> Parser state (Step e e)
-expressionHelp ((Config conf) as config) precedence leftExpression =
+expressionHelp ((Config conf) as config) currentPrecedence leftExpression =
     conf.spaces
         |> Combine.continueWith
             (Combine.oneOf
-                [ operation config precedence leftExpression
+                [ operation config currentPrecedence leftExpression
                     |> Combine.map Loop
                 , Combine.succeed (Done leftExpression)
                 ]
             )
 
 
 operation : Config state e -> Int -> e -> Parser state e
-operation ((Config conf) as config) precedence leftExpression =
+operation ((Config conf) as config) currentPrecedence leftExpression =
     conf.andThenOneOf
-        |> List.filterMap (\toOperation -> filter (toOperation config) precedence leftExpression)
-        |> Combine.oneOf
-
-
-filter : ( Int, e -> Parser state e ) -> Int -> e -> Maybe (Parser state e)
-filter ( precedence, parser ) currentPrecedence leftExpression =
-    if precedence > currentPrecedence then
-        Just (parser leftExpression)
+        |> List.filterMap
```

Comment:
I don't know how often this is called, but (as you know) most `List.` helpers aren't well optimized.
For e.g. `filterMap`, a fold or recursion version is twice as fast at least (more if you don't care about resulting order).
![image](https://github.com/user-attachments/assets/542caca7-6108-4f52-bbb7-8b8b127732d3)
```elm
listJustsToAnyOrderUsingRecursion : (element -> Maybe value) -> List element -> List value
listJustsToAnyOrderUsingRecursion elementToMaybe list =
    listJustsToReverseOrderUsingRecursionFrom [] elementToMaybe list


listJustsToReverseOrderUsingRecursionFrom : List value -> (element -> Maybe value) -> List element -> List value
listJustsToReverseOrderUsingRecursionFrom soFar elementToMaybe list =
    case list of
        [] ->
            soFar

        head :: tail ->
            listJustsToReverseOrderUsingRecursionFrom
                (case elementToMaybe head of
                    Nothing ->
                        soFar

                    Just value ->
                        value :: soFar
                )
                elementToMaybe
                tail
```

Reply:
You're right. In elm-review however, I patch these core functions to be the same as the one in elm-optimize-level-2, where they are fast.
If I use a custom solution, then I can't really patch unfortunately (and it will be slow-ish for all)

I don't know how much I should consider this not running through elm-review 😅

Reply:
> For e.g. filterMap, a fold or recursion version can twice as fast (more if you don't care about resulting order).

in this case, we care about the order because the most used parsers are on top. Although we could simply reverse the original list :)